messages/teleporter/message_handler.go

```diff
@@ -305,10 +305,32 @@ func (m *messageHandler) SendMessage(
 	}
 
 	// Wait for the message to be included in a block before returning
-	err = m.waitForReceipt(signedMessage, m.destinationClient, txHash, m.teleporterMessageID)
+	receipt, err := m.waitForReceipt(signedMessage, m.destinationClient, txHash, m.teleporterMessageID)
 	if err != nil {
 		return common.Hash{}, err
 	}
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		// Check if the message has already been delivered to the destination chain
+		teleporterMessenger := m.getTeleporterMessenger()
+		delivered, err := teleporterMessenger.MessageReceived(&bind.CallOpts{}, m.teleporterMessageID)
+		if err != nil {
+			m.logger.Error(
+				"Failed to check if message has been delivered to destination chain.",
+				zap.Error(err),
+			)
+			return common.Hash{}, fmt.Errorf("failed to check if message has been delivered: %w", err)
+		}
+		if delivered {
+			m.logger.Info("Execution reverted: message already delivered to destination.")
+			return txHash, nil
+		}
+
+		m.logger.Error(
```

Comment:
I know this log was already existing, but should we add the blockchain ID field to for easier debugging going forward?

Reply:
See [my comment here](https://github.com/ava-labs/icm-services/pull/759/files?diff=unified&w=1#r2057296450), the blockchainID is already included. I'm not sure if there's a great way to document this though.